packages/expo-system-ui/plugin/src/withIosStatusBarStyle.ts

```diff
@@ -0,0 +1,19 @@
+import { ConfigPlugin, withInfoPlist } from 'expo/config-plugins';
+
+export const withIosStatusBarStyle: ConfigPlugin = (config) =>
+  withInfoPlist(config, (config) => {
+    const { experiments = {}, userInterfaceStyle = 'automatic' } = config;
+    const { edgeToEdge = false } = experiments;
+
+    const styles = {
+      automatic: 'UIStatusBarStyleDefault',
+      dark: 'UIStatusBarStyleLightContent',
+      light: 'UIStatusBarStyleDarkContent',
+    };
+
+    config.modResults['UIStatusBarStyle'] = edgeToEdge
```

Comment:
Can you highlight how this works in iOS and why this config plugin is needed? Isn't edge to edge kind of "built in" in iOS?

Reply:
To avoid a system bars style blink on app start, and until the JS is parsed / booted, the style is overridden  using `userInterfaceStyle` value.

- `automatic` -> `dark-content` with light theme, `light-content` with dark theme
- `light` -> `dark-content`
- `dark` -> `light-content`

On Android, it sets `windowLightSystemBars` if needed, on iOS, it sets `UIStatusBarStyle`. On both platforms, the "gesture navigation bar" style is automatic.

Reply:
Ultimately, I'm working on handling automatically the system bars style on android, similar to the iOS behavior, as it's a way better DX. Here's [my WIP](https://twitter.com/zoontek/status/1797393598884847908/video/1).

Instead of `auto` just being "`light` in dark mode, `dark` in light mode", it will able to measure contrast as well. Instead of aligning both platforms to the android behavior, we will align both to the iOS one, but ideally, both will always behave the same.

Reply:
the only thing to look out for re: measuring contrast - i think we discussed this on our call - is that perhaps there will be implications for permissions in the future

Reply:
Ideally, I would love to discuss with a Google engineer / product about that. If they can makes its style automatic + provides an AndroidX compat lib based on this trick, that could be awesome.

But they are really hard to join.